### DISCUSSION TITLE
Pdf not shown correctly on windows

### DISCUSSION THREAD

**Discussion by KillTheMule** [typst/typst#2060] [General]
I have this pdf made with typst on linux. It does show correctly, here is a screenshot of a relevant part:

![screenshot_linux](https://github.com/typst/typst/assets/4117685/7ef7a2d3-7943-4daa-8d73-c7b4c66d8614)

This is how it is supposed to look like :) Opening the file on windows, however, the icons vanish (I tried edge, firefox, and sumatra pdf), like this (sorry, the shot is quite a bit smaller, but should show that all the icons are simply missing):

<img width="95" alt="screenshot_windows" src="https://github.com/typst/typst/assets/4117685/c59a68c4-82b0-419d-8137-aaf1f6c41e0f">

The whole thing is part of a table, the dots were made with `repeat`, and the icons are in individual columns, made with `sym.square.stroked.medium` (or the filled variant). One thing of note: On linux, mupdf shows this warning:

```
warning: Bad or missing last pointer in outline tree, repairing
```

I have not seen any consequence of this on linux, though, everything looks just fine.

So, why are the symbols missing on windows and what can I do? Sumatra pdf shows that the used fonts are embedded in the pdf (they're all latin modern fonts I'm explicitely using). I might get by replacing the icons by images as a workaround, but I'd appreciate not having to do that.

Thanks for any pointers! If I should provide more info, please let me know :)

**Comment by Enter-tainer** [comment]
https://github.com/typst/typst/issues/2032

possibly related

**Reply by Dherse** [reply]
After seeing OP's answer, it looks like that might indeed be the case

**Comment by Dherse** [comment]
Can you send the PDF either here or privately/on Discord, so that I can have a look? @LaurenzV @reknih and I are the most "up to date" and knowledgeable on PDF export I think so we would need to take a look at what typst generated.

**Comment by KillTheMule** [comment]
Compiling on windows produces a pdf that shows correctly on windows AND linux. Sumatra pdf lists an additional included font though, called `SegoeUIEmoji ` (the fonts are somewhat incorrect since I do not have latin modern installed on windows).

Now, running `pdffonts` on linux shows 4 fonts for the linux-produced pdf as well (sumatra pdf only listed 3  on windows), but throws an error `Syntax Warning: Mismatch between font type and embedded font file`.

@Dherse The pdf needs to stay private, but I'll produce a reduced version that inhibits this problem and send it via discord to you. I'll include a version produced on windows as well, so you can compare :)

**Reply by Dherse** [reply]
Could you also try (if you can build it that is) the latest build of typst, yesterday some changes to fonts were pushed to solve some issue, perhaps it might help

**Reply by laurmaedje** [reply]
If it doesn't help, it might be related to the new FIXME 2 indicated in the code: https://github.com/typst/typst/blob/c0b332a4c07b95c0c3196d0d7b149560dcef67fa/crates/typst/src/export/pdf/font.rs#L47

The linked Stack overflow answer also talks about a Han font albeit a different one. Maybe this one also uses CIDFont operators in its Top Dict.

**Reply by KillTheMule** [reply]
Ok, I've quite a journey behind me. It seems I have confused several pdf files, but to make sure I took several detours and tries. The defunct pdf was generated by my own code calling the typst library where I failed to include the necessary font (while investigating I stumbled on an (possible) oddity in font selection in typst, which I had to work around first). I had the file confused with ones generated by typst-cli, hence this question.

But: I can now report that both with the latest typst release as well as current HEAD, the fonts get included properly and render fine on windows as well as linux. The warning `Syntax Warning: Mismatch between font type and embedded font file` I mentioned only exists when using typst 0.7, so current HEAD seems to be an improvement indeed, although I could not discern a visible difference.

Thanks all for chiming in :) Love typst!
